tdp/cli/commands/stale/__init__.py

```diff
@@ -0,0 +1,136 @@
+
+    cluster_variables = ClusterVariables.get_cluster_variables(
+        collections=collections, tdp_vars=vars, validate=validate
+    )
+    check_services_cleanliness(cluster_variables)
+
+    session_class = get_session_class(database_dsn)
+    with session_class() as session:
+        if generate:
+            click.echo("Generating the list of stale components.")
+            deployed_component_version_logs = get_latest_success_component_version_log(
+                session
+            )
+            modified_services_or_components_names = cluster_variables.get_modified_components_names(
+                services_components_versions=deployed_component_version_logs
+            )
+            modified_components_names = set()
+            for modified_component_name in modified_services_or_components_names:
+                click.echo(f"{modified_component_name.full_name} has been modified.")
+                if modified_component_name.is_service:
+                    config_operations_of_the_service = filter(
+                        lambda operation: operation.action_name == "config"
+                        and not operation.is_service_operation(),
+                        dag.services_operations[modified_component_name.service_name],
+                    )
+                    modified_components_names.update(
+                        [
+                            operation.name
+                            for operation in config_operations_of_the_service
+                        ]
+                    )
+                else:
+                    modified_components_names.add(
+                        f"{modified_component_name.full_name}_config"
+                    )
+            operations = dag.get_operations(
+                sources=list(modified_components_names), restart=True
+            )
+            config_and_restart_operations = dag.filter_operations_regex(
+                operations, r".+_(config|(re|)start)"
+            )
+            components_status_dict = defaultdict(_ComponentStatus)
+            for operation in config_and_restart_operations:
+                if operation.is_service_operation():
+                    continue
+                component_status = components_status_dict[(operation.service_name, operation.component_name)]
+                if operation.action_name == "config":
+                    component_status.to_reconfigure = True
+                if operation.action_name == "restart":
+                    component_status.to_restart = True
+            stale_components = [
+                StaleComponent(
+                    service_name=service_name,
+                    component_name=component_name,
+                    to_reconfigure=component_status.to_reconfigure,
+                    to_restart=component_status.to_restart,
+                )
+                for (service_name, component_name), component_status in components_status_dict.items()
+            ]
```

Comment:
I think it will do the same if I understand correctly. This remove the use of class `_ComponentStatus`.
```suggestion
            stale_components = []
            for operation in config_and_restart_operations:
                if operation.is_service_operation():
                    continue
                stale_components.append(
                    StaleComponent(
                        service_name=operation.service_name,
                        component_name=operation.component_name,
                        to_reconfigure=operation.action_name == "config",
                        to_restart=operation.action_name == "restart",
                    )
                )
```